src/Compilers/Core/Portable/SourceGeneration/UserFunction.cs

```diff
@@ -25,6 +25,8 @@ internal sealed class WrappedUserComparer<T> : IEqualityComparer<T>
     {
         private readonly IEqualityComparer<T> _inner;
 
+        public static WrappedUserComparer<T> Default { get; } = new WrappedUserComparer<T>(EqualityComparer<T>.Default);
+
         public WrappedUserComparer(IEqualityComparer<T> inner)
```

Comment:
When you originally wrote this rather than a ctor did you consider the following? 

```csharp
public static WrappedUserComparer Create(IEqualityComparer<T> comparer) => 
  comparer is WrappedUserComparer<T> w ? w : new WrappedUserComparer(comparer);
```

One concern I had when looking through this code was if we were inadvertently creating nested wraps.

Reply:
It's an internal class and you can only create it via an extension method that doesn't return it, so it shouldn't be possible to actually get an instance of it that can be passed to the ctor as an external user.